client/network/test/src/sync.rs

```diff
@@ -657,3 +657,40 @@ fn full_sync_requires_block_body() {
 	net.block_until_idle();
 	assert_eq!(net.peer(1).client.info().best_number, 0);
 }
+
+#[test]
+fn imports_stale_once() {
+	let _ = ::env_logger::try_init();
+
+	fn import_with_announce(net: &mut TestNet, hash: H256) {
+		// Announce twice
+		net.peer(0).announce_block(hash, Vec::new());
```

Comment:
Why do we announce it twice?

Reply:
This is a test for the case when duplicate announcement lead to stale block being imported twice.